server/services/Transmission/clientGatewayService.ts

```diff
@@ -383,7 +383,7 @@ class TransmissionClientGatewayService extends ClientGatewayService {
           ...(await Promise.all(
             torrents.map(async (torrent) => {
               const percentComplete = (torrent.haveValid / torrent.totalSize) * 100;
-              const ratio = torrent.downloadedEver === 0 ? -1 : torrent.uploadedEver / torrent.downloadedEver;
+              const ratio = torrent.totalSize === 0 ? -1 : torrent.uploadedEver / torrent.totalSize;
```

Comment:
how could `torrent.totalSize === 0` ?

Reply:
It's should not happen, but if it does might as well handle it.

Reply:
Don't think this will happen. also it's wrong. we should use totalSize based ratio only when torrent is not downloaded at all, which means `torrent.downloadedEver === 0`, not `torrent.totalSize === 0`

Reply:
I see so something like this  ?

```ts
const ratio = torrent.downloadedEver === 0 ? -1 torrent.uploadedEver / torrent.totalSize : torrent.uploadedEver / torrent.downloadedEver;
```

Reply:
yes